lib/y2lan_restart_common.pm

```diff
@@ -148,15 +149,22 @@ sub check_network_status {
         assert_script_run 'dig suse.com|grep \'status: NOERROR\'';    # test if conection and DNS is working
     }
     assert_script_run 'cat journal.log';                              # print journal.log
-    my $is_empty = script_run('[ -s journal.log ]');
-    if ($is_empty) {
-        assert_equals($expected_status, 'no_restart', "Network restart didn't occur, even though it was expected.");
+    my $journal_findings = script_output('cat journal.log');
+    record_info "$expected_status", "Network status expects $expected_status";
+    if ($expected_status eq 'no_restart') {
+        assert_matches(qr/^\s*$/, $journal_findings, "Network service should not have been restarted/reloaded. Check journalctl findings \n$journal_findings\n\n");
+    }
+    elsif ($expected_status eq 'restart') {
+        assert_matches(qr/($query_pattern_for_restart)/, $journal_findings, "Network restart was expected. Check journalctl findings \n$journal_findings\n\n");
```

Comment:
minor thing: no need for parenthesis I think: `qr/$query_pattern_for_restart/`

Reply:
Actually i does. the $query_pattern_for_restart contains a `or` condition.

Reply:
does it fails the expansion/interpolation in that case?, it looks to me like an additional capture group no needed, but I might be wrong.

Reply:
apologies. it works even without. i though it doesnt

Reply:
Although i would rather leave it for now. but thanks. good to know

Reply:
if you already confirmed, I'm not sure why don't remove those 2 characters "now", with re-running one verification should be enough to see that all is as expected. The problem is that it creates a pattern in the repo that other contributors might think that is needed and that they have to follow (it is exactly what I did myself to verify it, search for it). Anyway is minor, feel free to close this conversation in that case.

Reply:
regex adjusted. i am resolving this